render-only-sample/rosumd/RosUmdDevice.cpp

```diff
@@ -2054,6 +2036,50 @@ void RosUmdDevice::RefreshPipelineState(UINT vertexOffset)
 
 #if VC4
 
+VC4TextureType RosUmdDevice::MapDXGITextureFormatToVC4Type(RosHwLayout layout, DXGI_FORMAT format)
+{   
+    VC4TextureType textureType;
+    textureType.TextureType = VC4_TEX_RGBA32R;
+
+    // Map texture layout and DXGI format to HW format
+    // Note: Some of the DXGI formats are emulated (for example, during 
+    // initialization, texture is converted to HW-compatible format)
+    if (layout == RosHwLayout::Tiled)
+    {
+        switch (format)
+        {
+            case DXGI_FORMAT_R8G8B8A8_UNORM:
+            {
+                textureType.TextureType = VC4_TEX_RGBX8888;
+            } 
+            break;
+            case DXGI_FORMAT_R8_UNORM:
+            {
+                textureType.TextureType = VC4_TEX_RGBX8888;
+            }
+            break;
+            case DXGI_FORMAT_R8G8_UNORM:
+            {
+                textureType.TextureType = VC4_TEX_RGBX8888;
+            }
+            break;
+            case DXGI_FORMAT_A8_UNORM:
+            {
+                textureType.TextureType = VC4_TEX_RGBX8888;
+            }
+            break;
+        }
```

Comment:
Consider adding default case and asserting so that we know when unimplemented functionality is hit.